waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -543,38 +560,62 @@
          :message "Error while scaling waiter service"})))
 
   (retrieve-directory-content
-    [{:keys [http-client] {:keys [port scheme]} :fileserver}
-     _ instance-id host browse-path]
-    (let [auth-str @k8s-api-auth-str
-          headers (when auth-str {"Authorization" auth-str})
-          instance-restart-number (->> (string/last-index-of instance-id \-)
-                                       inc
-                                       (subs instance-id))
-          instance-base-dir (str "/r" instance-restart-number)
+    [{:keys [http-client log-bucket-url service-id->service-description-fn watch-state]
+      {:keys [port scheme]} :fileserver}
+     service-id instance-id host browse-path]
+    (let [{:keys [_ pod-name restart-number]} (unpack-instance-id instance-id)
+          instance-base-dir (str "r" restart-number)
           browse-path (if (string/blank? browse-path) "/" browse-path)
           browse-path (cond->
                         browse-path
                         (not (string/ends-with? browse-path "/"))
                         (str "/")
                         (not (string/starts-with? browse-path "/"))
                         (->> (str "/")))
-          target-url (str scheme "://" host ":" port instance-base-dir browse-path)]
-      (when port
-        (ss/try+
-          (let [result (http-utils/http-request
-                         http-client
-                         target-url
-                         :accept "application/json"
-                         :content-type "application/json"
-                         :headers headers)]
-            (for [{entry-name :name entry-type :type :as entry} result]
-              (if (= "file" entry-type)
-                (assoc entry :url (str target-url entry-name))
-                (assoc entry :path (str browse-path entry-name)))))
-          (catch [:client http-client] response
-            (log/error "request to fileserver failed: " target-url response))
-          (catch Throwable t
-            (log/error t "request to fileserver failed"))))))
+          run-as-user (-> service-id
+                          service-id->service-description-fn
+                          service-description->namespace)
+          pod (get-in @watch-state [:service-id->pod-id->pod service-id pod-name])]
+      (ss/try+
```

Comment:
Nitpick: I would extract function for the two cases of getting the info from the sidecar and from S3.